openzwave-sys/build.rs

```diff
@@ -6,10 +31,29 @@ fn main() {
      .file("openzwave-c/manager.cc")
      .file("openzwave-c/notification.cc")
      .file("openzwave-c/value_classes/value_id.cc")
-     .include("/usr/include/openzwave") // workaround for https://github.com/OpenZWave/open-zwave/issues/771
-     .include("/usr/local/include/openzwave")
      .cpp(true)
-     .flag("-std=c++11") // to iterate with ranges
-     .compile("libopenzwave-c.a");
-    println!("cargo:rustc-flags=-l openzwave");
+     .flag("-std=c++11"); // to iterate with ranges
+
+    if find_include_path(&mut c).is_ok() {
+        c.compile("libopenzwave-c.a");
+        println!("cargo:rustc-flags=-l openzwave");
+        return;
+    }
+
+    // OK then let's build the real stuff...
+    println!("Building the open-zwave library...");
+    let exit_code = Command::new("make").current_dir("open-zwave").status().unwrap();
+    if !exit_code.success() {
+        panic!("Could not build the open-zwave library.");
+    }
+
+    // then build our library
+    c.include("open-zwave/cpp/src").compile("libopenzwave-c.a");
+
+    let this_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+
+    // then tell rust to link with open-zwave
+    println!("cargo:rustc-link-lib=udev");
```

Comment:
Yeah udev only exists on linux. So you'll probably need to have two separate function to do the println! (one for linux and one one for non-linux) and use:
#[cfg(target_os = "linux")] and #[cfg(not(target_os = "linux"))] in front of the 2 functions.